Exercises/NoModules/Chapter 03/Soln3_04.cpp

```diff
@@ -0,0 +1,49 @@
+// Exercise 3-4. Packing and unpacking characters.
+
+// Strictly speaking, whether or not this program works is compiler-dependent.
+// Although it's extremely unlike that you'll notice a problem.
+// Concretely: it'll only work if a single byte is 8 bit (virtually always the case),
+// and the fundamental type int counts at least 4 bytes (true for most modern systems).
+
+#include <iostream>
+#include <iomanip>
+
+int main()
+{
+  unsigned int packed {};
+  unsigned char ch {};
+  std::cout << std::left << std::setw(26) << "Enter a character: ";
+  std::cin >> ch;
+  packed |= ch;
+
+  std::cout << std::setw(26) << "Enter a second character: ";
+  std::cin >> ch;
+  packed <<= 8;                                // Shift left 1 byte
+  packed |= ch;
+
+  std::cout << std::setw(26) << "Enter a third character: ";
+  std::cin >> ch;
+  packed <<= 8;                                // Shift left 1 byte
+  packed |= ch;
+
+  std::cout << std::setw(26) << "Enter a fourth character: ";
+  std::cin >> ch;
+  packed <<= 8;                                // Shift left 1 byte
+  packed |= ch;
+
+  std::cout << "The word containing 4 packed characters is " << std::right
```

Comment:
This would be nice example to use std::format().

Reply:
Especially since the reader does not know std::setw()!